vendor/github.com/openshift/library-go/pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -47,6 +47,36 @@ type InstallerController struct {
 
 	// installerPodImageFn returns the image name for the installer pod
 	installerPodImageFn func() string
+
+	// installStateFunc derives the static pod install state of a node
+	installStateFunc StaticPodInstallStateFunc
+}
+
+// StaticPodInstallState is the status of a static pod installation to a node.
+type StaticPodInstallState int
+
+const (
+	// StaticPodInstallStateRunning means that the static pod installation of a node is running.
+	StaticPodInstallStateRunning = StaticPodInstallState(iota)
+	// StaticPodInstallStateSuccess means that the static pod installation of a node has succeeded.
+	StaticPodInstallStateSuccess
+	// StaticPodInstallStateSuccess means that the static pod installation of a node has failed.
+	StaticPodInstallStateFailed
+)
+
+// StaticPodInstallStateFunc derives the status of the installer.
+type StaticPodInstallStateFunc func(installerPod *corev1.Pod, deploymentID int32, nodeName string) (StaticPodInstallState, error)
```

Comment:
Given a nodeName, we could return the current deployment and it's state, right?  That seems generally useful to building levels and would also satisfy the need here.  You're waiting for it to return the "correct" deploymentID and the running state.